### PR TITLE
Commit table content if user wants to

### PR DESCRIPTION
**Pull request by NicolasBusson**
Hello,
To make it possible to import and commit database content if the user confirms that's what he wants to in the popup, I had to change a few things during /data/ files (de)serialization:

- Fixed error message "Object type TABU is not supported by this system". To avoid this error I filtered objects coming from /data/ folder in ZCL_ABAPGIT_FILE_DESERIALIZE->FILTER_FILES_TO_DESERIALIZE
- Fixed fields with date format that always appear as different between local and remote repository because they are not formatted exactly the same way during serialization and deserialization. See line lo_ajson->format_datetime( ) added to ZCL_ABAPGIT_DATA_SERIALIZER->CONVERT_ITAB_TO_JSON

Hope that helps!
Cheers,
Nick

### CODE REVIEW

File: src/repo/zcl_abapgit_repo.clas.abap [hunk 1]
```diff
@@ -262,7 +262,11 @@ CLASS ZCL_ABAPGIT_REPO IMPLEMENTATION.
 
     DATA: lt_updated_files TYPE zif_abapgit_definitions=>ty_file_signatures_tt,
           lt_result        TYPE zif_abapgit_data_deserializer=>ty_results,
-          lx_error         TYPE REF TO zcx_abapgit_exception.
+          lx_error         TYPE REF TO zcx_abapgit_exception,
+          ls_result        TYPE zif_abapgit_data_deserializer=>ty_result,
+          ls_overwrite     TYPE zif_abapgit_definitions=>ty_overwrite.
+
+    FIELD-SYMBOLS <tab> TYPE ANY TABLE.
```
Comment (larshp) [inline src/repo/zcl_abapgit_repo.clas.abap:269]: as indicated by abaplint, please rename this variable to something like `<lt_tab>`

**Comment by mbtools** [comment]
Hi Nick,

ajson mirror is scheduled to run again tomorrow. this will take care of the ajson bug and also the date formatting (since it's the default with the new ajson version).

updating the DB tables is already coded. after `deserialize` you would just have to call `actualize`:

```
    lt_result = zcl_abapgit_data_factory=>get_deserializer( )->deserialize(
      ii_config  = get_data_config( )
      it_files   = get_files_remote( ) ).

    zcl_abapgit_data_factory=>get_deserializer( )->actualize( lt_result ).
```

We did not include this in abapGit yet since there are concerns that the wrong tables could get update and therefore corrupt the target system. 

The discussion is at #3441 (start at https://github.com/abapGit/abapGit/issues/3441#issuecomment-804338583). Let's come to some conclusions there first before we implement something.

**Comment by NicolasBusson** [comment]
Hi Marc,
Thanks a lot for the hint! I had read discussion #3441 and saw there were some concerns about table updates... but I confess I didn't see this method "actualize" existed to commit content into database tables... I'm sorry.
I just gave it a look and it seems you do not take into account IS_CHECKS-OVERWRITE with the user decision to actually go ahead with the modification. But maybe this is done somewhere else. So here is what I suggest: I'll wait until tomorrow to get the latest version of ajson and abapgit on my system. Then I'll add the call to actualize method as you suggested above and make some more tests. Sounds like a plan? And in the meantime do you think I should I close this pull request then?
And thanks again for everything that you do.
Cheers,
Nick

**Comment by mbtools** [comment]
Close this and open a new PR for the `zcl_abapgit_file_deserialize` fix which is a bug. Thanks for finding it.

Let's continue in #3441

**Pull request closed without merging**